actors/builtin/miner/miner_actor.go

```diff
@@ -475,6 +477,9 @@ func (a Actor) PreCommitSector(rt Runtime, params *SectorPreCommitInfo) *adt.Emp
 	store := adt.AsStore(rt)
 	var st State
 	newlyVestedAmount := rt.State().Transaction(&st, func() interface{} {
+		vestingFunds, err := st.LoadVestingFunds(store)
```

Comment:
As commented below, this is less convenient but no more efficient, since there is exactly one mutation between the load and store. Push this back down into `UnlockVestedFunds`.